jfxtras-agenda/src/main/java/jfxtras/internal/scene/control/skin/agenda/base24hour/AppointmentAbstractTrackedPane.java

```diff
@@ -27,322 +27,322 @@
+				// if it does not overlap, we start a new cluster
+				if (lOverlaps == false) {
+					lClusterOwner = lAppointmentPane;
+					lClusterOwner.clusterMembers = new ArrayList<AppointmentAbstractTrackedPane>(); 
+					lClusterOwner.clusterTracks = new ArrayList<List<AppointmentAbstractTrackedPane>>();
+					lClusterOwner.clusterTracks.add(new ArrayList<AppointmentAbstractTrackedPane>());
+				}
+			}
+			
+			// add it to the track (and setup all other cluster data)
+			lClusterOwner.clusterMembers.add(lAppointmentPane);
+			lClusterOwner.clusterTracks.get(lTrackNr).add(lAppointmentPane);
+			lAppointmentPane.clusterOwner = lClusterOwner;
+			lAppointmentPane.clusterTrackIdx = lTrackNr;				
+			// for debug  System.out.println("----"); for (int i = 0; i < lClusterOwner.clusterTracks.size(); i++) { System.out.println(i + ": " + lClusterOwner.clusterTracks.get(i) ); } System.out.println("----");
+		}
+		
+		// done
+		return appointmentAbstractTrackedPanes;
+	}
+	
+	/**
+	 * 
+	 */
+	static private int determineTrackWhereAppointmentCanBeAdded(List<List<AppointmentAbstractTrackedPane>> tracks, AppointmentAbstractTrackedPane appointmentPane)
+	{
+		int lTrackNr = 0;
+		while (true)
+		{
+			// make sure there is a arraylist for this track
+			if (lTrackNr == tracks.size()) {
+				tracks.add(new ArrayList<AppointmentAbstractTrackedPane>());
+			}
+			
+			// scan all existing appointments in this track and see if there is an overlap
+			if (checkIfTheAppointmentOverlapsAnAppointmentAlreadyInThisTrack(tracks, lTrackNr, appointmentPane) == false)
+			{
+				// no overlap, it can be added here
+				return lTrackNr;
+			}
+
+			// overlap, try next track
+			lTrackNr++;
+		}
+	}
+	
+	/**
+	 * 
+	 */
+	static private boolean checkIfTheAppointmentOverlapsAnAppointmentAlreadyInThisTrack(List<List<AppointmentAbstractTrackedPane>> tracks, int tracknr, AppointmentAbstractTrackedPane newAppointmentPane)
+	{
+		// get the track
+		List<AppointmentAbstractTrackedPane> lTrack = tracks.get(tracknr);
+		
+		// scan all existing appointments in this track
+		for (AppointmentAbstractTrackedPane lAppointmentPane : lTrack)
+		{
+			// There is an overlap:
+			// if the start time of the already placed appointment is before or equals the new appointment's end time 
+			// and the end time of the already placed appointment is after or equals the new appointment's start time
+			// ...PPPPPPPPP...    
+			// .NNNN.......... -> Ps <= Ne & Pe >= Ns -> overlap
+			// .....NNNNN..... -> Ps <= Ne & Pe >= Ns -> overlap
+			// ..........NNN.. -> Ps <= Ne & Pe >= Ns -> overlap
+			// .NNNNNNNNNNNNN. -> Ps <= Ne & Pe >= Ns -> overlap
+			// .N............. -> false    & Pe >= Ns -> no overlap
+			// .............N. -> Ps <= Ne & false	  -> no overlap
+			LocalDateTime lPlacedStart = lAppointmentPane.startDateTime;
+			LocalDateTime lPlacedEnd = (lAppointmentPane.endDateTime != null ? lAppointmentPane.endDateTime.minusSeconds(10) : lAppointmentPane.startDateTime.plusMinutes(10));
```

Comment:
If you subtract 10 min from the end, you change the rules from Pe >= Ns to Pe-10 min >= Ns. This change would cause a false negative in the following scenario

New appointment is from 8:05am - 8:25am
Existing appointment is from 7:10am - 8:10am

This change would make the existing appointment end date to be 8:00am which would mean there is no overlap even though there is.

Are there JUnits for the commented scenarios?